dashboard/test/ui/features/step_definitions/pd.rb

```diff
@@ -214,6 +222,236 @@
+        order: 4,
+        hidden: true
+      }
+    ].to_json
+  )
+
+  Pd::SurveyQuestion.find_or_create_by!(form_id: CDO.jotform_forms['local']['facilitator'],
+    questions: [
+      {
+        id: 1,
+        name: 'textarea',
+        type: 'textarea',
+        text: 'How was your facilitator?',
+        order: 1
+      },
+      {
+        id: 2,
+        name: 'userId',
+        text: 'userId',
+        type: 'textarea',
+        order: 2,
+        hidden: true
+      },
+      {
+        id: 3,
+        name: 'workshopId',
+        text: 'workshopId',
+        type: 'textarea',
+        order: 3,
+        hidden: true
+      },
+      {
+        id: 4,
+        name: 'sessionId',
+        text: 'sessionId',
+        type: 'textarea',
+        order: 4,
+        hidden: true
+      },
+      {
+        id: 5,
+        name: 'facilitatorId',
+        text: 'facilitatorId',
+        type: 'textarea',
+        order: 5,
+        hidden: true
+      },
+      {
+        id: 6,
+        name: 'day',
+        text: 'day',
+        type: 'textarea',
+        order: 6,
+        hidden: true
+      }
+    ].to_json
+  )
+
+  10.times do |x|
+    user = workshop.enrollments[x].user
+
+    Pd::WorkshopDailySurvey.create!(
+      form_id: CDO.jotform_forms['local']['day_0'],
+      submission_id: Pd::WorkshopDailySurvey.maximum(:submission_id) + 1,
+      pd_session: nil, #No session for the first survey
+      answers: {
+        '1': {
+          'I like computer science': ['Strongly Agree', 'Agree', 'Slightly Agree'][x % 3],
+          'People should learn computer science': ['Strongly Agree', 'Agree'][x % 2],
+          'I feel like I can teach computer science': ['Strongly Agree', 'Agree', 'Disagree'][x % 3]
+        },
+        '2': (1..5).to_a[x % 5].to_s,
+        '3': ['None', 'Some basic messing around', 'Formal education', 'I am l33t h4xx0r'][x % 4],
+        '4': ['Bill Nye', 'Ada Lovelace', 'Hadi', 'Hour of Code', 'Dunno'][x % 5],
+        '5': user.id,
+        '6': workshop.id,
+      }.to_json
+    )
+
+    Pd::WorkshopDailySurvey.create!(
+      form_id: CDO.jotform_forms['local']['day_1'],
+      submission_id: Pd::WorkshopDailySurvey.maximum(:submission_id) + 1,
+      user: workshop.enrollments[x].user,
+      pd_session: workshop.sessions.first,
+      pd_workshop: workshop,
+      answers: {
+        '1': %w(Amazing Brilliant Great Decent)[x % 4],
+        '2': user.id,
+        '3': workshop.id,
+        '4': workshop.sessions.first.id
+      }.to_json
+    )
+
+    Pd::WorkshopFacilitatorDailySurvey.create!(
+      form_id: CDO.jotform_forms['local']['facilitator'],
+      submission_id: Pd::WorkshopDailySurvey.maximum(:submission_id) + 1,
+      answers: {
+        '1': %w(Helpful Hillarious Inspiring Brilliant)[x % 4],
+        '2': user.id,
+        '3': workshop.id,
+        '4': workshop.sessions.first.id,
+        '5': workshop.facilitators[x % 2].id,
+        '6': 1
+      }.to_json
+    )
+  end
```

Comment:
This is a massive step. What about pulling out all the data creation code to a separate private method for readability?

Reply:
Sure thing